common-messaging/src/main/scala/org/genivi/sota/messaging/kafka/MessageListener.scala

```diff
@@ -15,33 +15,44 @@ object MessageListener {
 
   type MsgParser[T <: BusMessage] = T => Future[_]
 
-  def buildSource[T <: BusMessage](fromSource: Source[CommittableMessage[Array[Byte], T], NotUsed],
+  /**
+    * This trait is used to enable NATS to pretend it can do committable messages
+    */
+  trait CommittableMsg[T <: BusMessage] {
+    def msg(): T
+    def commitRecord(): Future[T]
+  }
+
+  class NatsMsg[T <: BusMessage](message: T) extends CommittableMsg[T] {
+    override def msg(): T = message
+
+    override def commitRecord(): Future[T] = Future.successful(msg)
+  }
+
+  class KafkaMsg[T <: BusMessage](message: CommittableMessage[_, T]) extends CommittableMsg[T] {
+    override def msg(): T = message.record.value()
+
+    override def commitRecord(): Future[T] = { message.committableOffset.commitScaladsl();Future.successful(msg()) }
```

Comment:
That's a Future[Done], which throws a type error as it can't be cast to T. Also, I don't see how that would allow me to forgo the map(_ => committableMsg)?